redash/tasks/queries/execution.py

```diff
@@ -111,10 +109,10 @@ def enqueue_query(query, data_source, user_id, is_api_key=False, scheduled_query
             if job_id:
```

Comment:
I know I said I’ll port `enqueue_query` from Celery to RQ as is and deal with refactoring later, but I’m really tempted to use RQ’s ability to set custom IDs to jobs and get rid of the whole lock mechanism using it (i.e. job IDs will be rq:job:<ds.id>:<query_hash> ). WDYT @arikfr?

Reply:
This is indeed tempting! But -- 

1. We lose ability to track down specific invocation (in logs and such), because every invocation has the same id.
2. Current hash implementation needs fixing (#2137).
3. Are we sure job IDs were meant to be used this way?

Reply:
1. That could be easily solved by a custom description or meta attribute holding an invocation counter.
2. That needs fixing regardless, but does that affect whether we should drop locks?
3. It's not a hack, if that's what you mean. RQ docs specify that you can use a predetermined ID instead of the auto-generated one.

Reply:
> RQ docs specify that you can use a predetermined ID instead of the auto-generated one.

But do they expect the ID to be non unique/reusable? An execution's id feels like something that supposed to be unique, I'm just concerned about the can of worms we're opening if we assign a non unique one.

Reply:
All jobs scheduled from `rq-scheduler` reuse the same identifier :)

Reply:
You might want to check out: https://github.com/rq/rq/issues/793.

Reply:
TL;DR: 

> What sirex said, the job will be executed twice (if the job is still in Redis).

Reply:
I fail to see the issue. We still check for the job's existence in Redis. We get the same effect of a lock by checking if the job is queued.

Reply:
I thought you were suggesting reusing job ids to avoid having to implement our own lock. 

While we don't need to worry about another key, we do need to worry about lots of other things and fully understand how rq works. Considering that the current implementation works and we have reasonable processes for it, I don't think we will gain much here at this stage, until we're more familiar with RQ. The risk vs. gain is just not worth it.